src/lamp_py/performance_manager/alerts.py

```diff
@@ -0,0 +1,478 @@
+    """
+
+    # pull out the active period timestamps from the dict in that column
+    def extract_start_end(
+        periods: Optional[List[Dict[str, int]]]
+    ) -> Tuple[int | None, int | None]:
+        """small lambda for extracting start and end timestamps"""
+        if periods is None:
+            return None, None
+        return periods[0].get("start"), periods[0].get("end")
+
+    alerts[
+        [
+            "active_period_start_timestamp",
+            "active_period_end_timestamp",
+        ]
+    ] = (
+        alerts["active_period"].apply(extract_start_end).apply(pandas.Series)
+    )
+
+    alerts = alerts.drop(columns=["active_period"])
+
+    # convert all of the timestamp columns to eastern standard time
+    timestamp_columns = [
+        "created",
+        "last_modified",
+        "last_push_notification",
+        "closed",
+        "active_period_start",
+        "active_period_end",
+    ]
+
+    for key in timestamp_columns:
+        timestamp_key = f"{key}_timestamp"
+        alerts[key] = (
+            pandas.to_datetime(alerts[timestamp_key], unit="s", utc=True)
+            .dt.tz_convert("America/New_York")
+            .dt.floor("s")
+        )
+
+    return alerts
+
+
+def explode_alerts(alerts: pandas.DataFrame) -> pandas.DataFrame:
+    """
+    the 'informed_entity' column is a list of dicts describing stops
+    along routes in directions that are effected by the alert. explode each
+    record along this list and extract the required information into new
+    columns
+    """
+    alerts = alerts.explode("informed_entity")
+
+    informed_entity_keys = [
+        "route_id",
+        "route_type",
+        "direction_id",
+        "stop_id",
+        "facility_id",
+        "activities",
+    ]
+
+    # extract information from the informed entity
+    for key in informed_entity_keys:
+        alerts[f"informed_entity.{key}"] = alerts["informed_entity"].apply(
+            lambda x, k=key: x.get(k)
+        )
+
+    alerts = alerts.drop(columns=["informed_entity"])
+
+    # transform the activities field from a list to a pipe delimitated string
+    alerts["informed_entity.activities"] = alerts[
+        "informed_entity.activities"
+    ].apply(lambda x: "|".join(str(item) for item in x if item is not None))
+
+    # the commuter rail informed entity contains extra details that aren't
+    # extracted in this transformation. those instances will appear as
+    # duplicated records, so remove them.
+    alerts = alerts.drop_duplicates()
+
+    return alerts
+
+
+def get_unprocessed_alert_files(
+    md_db_manager: DatabaseManager,
+) -> List[Dict[str, str]]:
+    """
+    Get unprocessed RT Alert Files from the MetadataLog table.
+
+    @return [
+        { 'pk_id': 1, 'path': 'lamp/RT_ALERTS/year=2024/month=1/day=4/hour=0/uuid_1-0.parquet' },
+        { 'pk_id': 2, 'path': 'lamp/RT_ALERTS/year=2024/month=1/day=4/hour=1/uuid_2-0.parquet' },
```

Comment:
these return paths would have the bucket_name in front of `lamp/`

Reply:
youre right, fixed.